src/App.tsx

```diff
@@ -1,61 +1,121 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+import { Todo } from './react-app-env';
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [selectedUserId, setSelectedUser] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const clearForm = () => {
+    setNewTodoTitle('');
+    setSelectedUser(0);
+  };
+
+  const generateId = () => {
+    let maxId = -Infinity;
+
+    todos.forEach(todo => {
+      if (todo.id > maxId) {
+        maxId = todo.id;
+      }
+    });
+
+    return maxId + 1;
+  };
+
+  const handleSubmitForm = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const newTodo = {
+      id: generateId(),
+      title: newTodoTitle,
+      completed: false,
+      userId: selectedUserId,
+    };
+
+    if (newTodoTitle.trim() && selectedUserId) {
+      setTodos(currentTodos => [...currentTodos, newTodo]);
+      clearForm();
+    }
+
+    if (!newTodoTitle.trim()) {
+      setHasTitleError(true);
+    }
+
+    if (!selectedUserId) {
+      setHasUserIdError(true);
+    }
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form onSubmit={handleSubmitForm}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            value={newTodoTitle}
+            placeholder="Enter todo"
+            onChange={event => {
+              const inputValue = event.target.value;
+
+              setNewTodoTitle(inputValue);
+
+              if (inputValue.trim().length <= 0) {
+                setHasTitleError(true);
+              } else {
+                setHasTitleError(false);
+              }
+            }}
```

Comment:
looks like the logic inside the click handler is too complicated to be inline. Can we make it a separate method?